boformer/NetworkSkins
# Pipet tool hovering

Fixes: Pippet tool sometimes cannot select roads.
### How to reproduce the problem:
- sub to [this city ](https://steamcommunity.com/sharedfiles/filedetails/?id=2022442822)
- use NS2 from workshop (it automatically loads when you sub to the mod)
- in the content manager select the game save and press sub to all.
- load the saved game
- try to hover over different roads in the city. In some areas of the city roads will not get a highlight.

Changes:
- One functional change (apart from the bug fix) is that the pippet cursor sprite appears only when the user hovers over segment as a visual feedback. when the user is not hovering over a segment then the cursor sprite becomes the default sprite.
- I copy pasted code from TMPE (with some improvements)
- Simulation step shall not deal with hovering. I moved the segment hovering functionality to OnToolUpdate which is part of the main thread.
- I copied code from TMPE (with some improvements).
- removed/simplified a bunch of redundant fields/properties.

Notes:
- Near the node its hard to select the segments you want see [#576](https://github.com/CitiesSkylinesMods/TMPE/issues/576) That is why I try to raycast node and get the segment from node.
- in the case of level crossings, I did not manage to raycast node. so I raycast segment, then get the closest node, then get the closest segment of that node again. this is necessary to fix  [#576](https://github.com/CitiesSkylinesMods/TMPE/issues/576) for level crossings.
